solsp/lsp/protocol.h

```diff
@@ -0,0 +1,678 @@
+#pragma once
+
+#include <json/value.h>
+
+#include <any>
+#include <array>
+#include <optional>
+#include <map>
+#include <string>
+#include <variant>
+#include <vector>
+
+// NOTE: https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/
+
```

Comment:
Did you have a look at https://reviews.llvm.org/diffusion/L/browse/clang-tools-extra/trunk/clangd/Protocol.h and https://reviews.llvm.org/diffusion/L/browse/clang-tools-extra/trunk/clangd/Protocol.cpp ?

Reply:
They basically translate the JSON to C++ structs before handling the requests. Isn't it what you both voted against? (I preferred that one). Or did you change your mind?

Reply:
Well, their files look like they could almost be taken over 1:1 - if that's possible, then we could still consider that. I'm mainly voting against writing something like that from scratch :-). But basically using their implementation might be worth exploring.

Reply:
I'm fine with copy'n'pasting the types from clangd, but I'm not a licensing expert (MIT vs GPL-2), I think we'd have to state that somewhere in our LICENSE.txt then, too?

Let me have a closer look at those next Monday afternoon (maybe you're in the office then, too?) Cheers. ;-)

Reply:
It looks to me to be too deeply tangled into the llvm codebase, so probably not worth it.